scripts/generate/templates/Heap.opts.js

```diff
@@ -0,0 +1,13 @@
+const makeType = (valueSize, indexSize) => ({
+  struct: `Uint${valueSize}Heap`,
+  node: `Uint${valueSize}HeapNode`,
+  valueSize,
+  valueType: `uint${valueSize}`,
+  indexSize,
+  indexType: `uint${indexSize}`,
+  blockSize: Math.ceil((valueSize + 2 * indexSize) / 256),
+});
+
+module.exports = {
+  TYPES: [makeType(256, 32), makeType(208, 24)],
```

Comment:
I dont think its realistic to push 4.2 billion values without poping any in the meantime. That would be 8.4 billions storage slot stores that are written to. That is the equivalent of 180 000 eth worth of gas (considering a very con servative 1 gwei/gas) ... just for writting to the data to the heap (not even including the rebalancing of the heap (and the other operations done at the same time). If you include the gas cost of all the reordering along the way, you are probable closer to 300 000 eth consumed on gas.

Note that if you heap has ~ 2**32 items already, pushing a new one means shitfing up ~32 levels of the tree. Each level requires reading the index + the value, which are 2 slots in the case of uint256. That is ~134k of gas just for warlming the slots. If you add the writes to reorder on top of that ... and its JUST for one structure push when the structure is close to full.

The idea was that both version use the same type for length so that the dev experience is similar, and IMO the limits are way sufficient.

Reply:
Right the numbers are huge, but imo is not unthinkable in the context of high performance L2s (for e.g. gaming)

Regardless, I agree that it's an edge case and I'd rather wait for a user to provide us feedback